doc/instances.md

```diff
@@ -402,6 +403,26 @@ ipv4.routes             | string    | -                 | no        | Comma deli
 ipv6.routes             | string    | -                 | no        | Comma delimited list of IPv6 static routes to add on host to nic
 boot.priority           | integer   | -                 | no        | Boot priority for VMs (higher boots first)
 
+#### nictype: ovn
```

Comment:
So I'm a bit confused by that one. I would expect a `nictype=ovn` NIC to connect to an OVN network which is not managed by LXD.

For anything managed by LXD, I would expect no `nictype` and juse use `network=` to point to the network, from that we can tell it's ovn and do the right thing.

I think the doc is a bit confusing as it has no section for the "no nictype" case where this would apply.
So maybe instead of adding `ovn` as a weird additional nictype, we should instead add a "no nictype" section which shows what can be used when `network=` is used, drop the `network` property from the `nictype: bridged` section and make sure that the keys we accept are all correct in documentation.

Reply:
@stgraber ah yes, I had missed adidng the ovn nic type docs when I added ovn networking support, so am filling this in now.

But whilst there is a "nic type" of ovn, you're right its misleading, as there is no way to specify it directly.

I'm thinking rather than split the different NIC types into different section groupings for when specifying `nictype` and when specifying `network`, rather keep those scenarios together in one section per-NIC type, but split the table in that section showing what can be specified with nictype only/both nictype and network. I feel this would reduce the duplication.

Then drop the "nictype" bit from the sub menu so it just says "ovn" or "bridge" in the sub heading to avoid confusion.

What do you think?

Reply:
I wouldn't have split them all, instead I would just have made a separate section for when no nictype is specified.
So basically:
 - network
 - nictype: physical
 - nictype: bridged
 - nictype: macvlan
 - nictype: ipvlan
 - nictype: p2p
 - nictype: sriov
 - nictype: routed

With the `network` one explaining that this applies to connecting to LXD managed networks, show the keys valid in that case with a column to indicate what network driver is needed for the given key.

That more closely matches with what our users experience as they either use nictype=FOO and parent=BAR or they use no nictype at all and use network=

Reply:
@stgraber hrm I see, although that approach has the disadvantage that it starts to go back to the previous way we structured this page with one big table for all nic types. Albeit it not the same extreme.

However it presents some issues in my view:

1. If the same key exists in different device types on different managed networks but with slightly different meaning we can't clearly explain the difference. Example off the top of my head is ipv4.routes, this has a different behaviour for bridge networks compared to physical networks. The former creates routes, the latter just allows them to be used by NICs. 

2. It makes finding info about a particular network device type more difficult because some nic devices are documented on a per type basis and others are merged together into a single table.

We could add an extra column to each nic device type's table called 'managed' indicating if the setting was available when linked to a managed network (this column would only be present for devices that supported managed networks, and we could add a 'supported modes' bit next to supported instance types listing which modes it supports. Along with a single top level section explaining the concept.

Reply:
Yeah, the part that's still messy with this is that as a user of `lxc config device add blah eth0 nic network=abc`, I may not know that `abc` is SR-IOV or macvlan or ovn and since I've never specified `nictype=`, I'm unlikely to go read up a `nictype` specific section in documentation.

Reply:
@stgraber ok how about this, I add the top-level network section as you mentioned earlier, listing the general keys that are available for NICs linked to managed networks irrespective of network type. This allows users who don't know the type of network they are connecting to to see how to do it and what is generally available.

Then in that section adding a list of links to the specific nic types that support managed networks (although will drop the prefix "nictype:" to avoid confusion) and then in those sections add a column listing which fields are usable when linked to a managed network (which will cover any specific settings or meanings for that network type).

Reply:
@stgraber I've updated the docs with what I have in mind, see what you think: 

https://github.com/lxc/lxd/pull/8031/commits/d4036eb6c26f7dc47ceeaf314c98c8c541d595f4